### ISSUE TITLE
On older Android devices, it looks like tracking is stopped at stopTimeout, even with disableStopDetection=true

### ISSUE THREAD

**Issue by adrianaioanei** [transistorsoft/react-native-background-geolocation#1093]
## Your Environment
* Plugin version: 3.7.0
* Platform: Android
* OS version: 5.0.2
* Device manufacturer / model: Sony Xperia
* React Native version (`react-native -v`): 0.61.5
* Plugin config
```javascript <!-- Syntax-highlighting:  paste your code below -->
const config: Config = {
  reset: true,
  desiredAccuracy: isPlatformAndroid ?
    BackgroundGeolocation.DESIRED_ACCURACY_HIGH :
    BackgroundGeolocation.DESIRED_ACCURACY_NAVIGATION,
  distanceFilter: 0,
  disableElasticity: true,
  stopOnTerminate: true,
  heartbeatInterval: 1, // in seconds
  disableStopDetection: true,
  stopOnStationary: false,
  foregroundService: true,
  debug: false,
  logLevel: BackgroundGeolocation.LOG_LEVEL_WARNING,
  logMaxDays: 2,
  // iOS:
  locationAuthorizationRequest: 'Always',
  showsBackgroundLocationIndicator: true,
  stationaryRadius: 1,
  preventSuspend: true,
  batchSync: true,
  maxBatchSize: 30,
  // Android:
  allowIdenticalLocations: true,
  locationUpdateInterval: 333,
  fastestLocationUpdateInterval: 333,
  activityRecognitionInterval: 0,
  notification: {
    text: I18n.t('text_notification_tracking'),
  },
}
```

## Expected Behavior
Tracking should continue after stopTimeout

## Actual Behavior
It looks like tracking stops after the default stopTimeout of 5 minutes. Even if that should not count since disableStopDetection is true.

## Context
Use the module for sail boat race tracking

## Debug logs

<details><summary>Logs</summary>

``` <!-- Syntax highlighting:  DO NOT REMOVE -->
D/TSLocationManager( 3689): ╔═════════════════════════════════════════════
D/TSLocationManager( 3689): ║ TrackingService: LocationResult
D/TSLocationManager( 3689): ╠═════════════════════════════════════════════
D/TSLocationManager( 3689): ╟─ ??????  Location[fused 54.341210,10.135730 acc=10 et=+9h30m10s768ms alt=89.0 vel=1.5298027E-6 bear=0.0 {Bundle[mParcelledData.dataSize=52]}], age: 66ms, time: 1593554523000
W/TSLocationManager( 3689): [c.t.l.service.TrackingService b] 
W/TSLocationManager( 3689):   ⚠️  performStopDetection found mStoppedAtLocation == null
D/TSLocationManager( 3689): [c.t.l.l.TSLocationManager onLocationResult] 
D/TSLocationManager( 3689): ╔═════════════════════════════════════════════
D/TSLocationManager( 3689): ║ Process LocationResult
D/TSLocationManager( 3689): ╠═════════════════════════════════════════════
D/TSLocationManager( 3689): [c.t.l.l.TSLocationManager calculateMedianAccuracy] Median accuracy: 9.0
I/TSLocationManager( 3689): [c.t.l.data.sqlite.b persist] 
I/TSLocationManager( 3689):   ✅  INSERT: a61c03c1-c1a8-4f6b-8e15-3735ee4b7403
I/TSLocationManager( 3689): [c.t.l.http.HttpService flush] 
I/TSLocationManager( 3689): ╔═════════════════════════════════════════════
I/TSLocationManager( 3689): ║ HTTP Service (count: 1)
I/TSLocationManager( 3689): ╠═════════════════════════════════════════════
I/TSLocationManager( 3689): [c.t.l.u.BackgroundTaskManager onStartJob] ⏳ startBackgroundTask: 787
D/TSLocationManager( 3689): [c.t.l.data.sqlite.b allWithLocking] 
D/TSLocationManager( 3689):   ✅  Locked 1 records
I/TSLocationManager( 3689): [c.t.l.http.HttpService a] 
I/TSLocationManager( 3689):   ??????  HTTP POST batch (1)
I/TSLocationManager( 3689): [c.t.l.http.HttpService$i onResponse] 
I/TSLocationManager( 3689):   ??????  Response: 200
D/TSLocationManager( 3689): [c.t.l.data.sqlite.b destroyAll] 
D/TSLocationManager( 3689):   ✅  DELETED: (1)
D/TSLocationManager( 3689): [c.t.l.data.sqlite.b allWithLocking] 
D/TSLocationManager( 3689):   ✅  Locked 0 records
I/TSLocationManager( 3689): [c.t.l.u.BackgroundTaskManager$Task stop] ⏳ stopBackgroundTask: 787
D/TSLocationManager( 3689): [c.t.l.s.BackgroundTaskService onDestroy] 
D/TSLocationManager( 3689): [c.t.l.service.TrackingService b] 
D/TSLocationManager( 3689): ╔═════════════════════════════════════════════
D/TSLocationManager( 3689): ║ TrackingService: LocationResult
D/TSLocationManager( 3689): ╠═════════════════════════════════════════════
D/TSLocationManager( 3689): ╟─ ??????  Location[fused 54.341210,10.135730 acc=10 et=+9h30m11s767ms alt=89.0 vel=0.0031872846 bear=180.0 {Bundle[mParcelledData.dataSize=52]}], age: 65ms, time: 1593554524000
W/TSLocationManager( 3689): [c.t.l.service.TrackingService b] 
W/TSLocationManager( 3689):   ⚠️  performStopDetection found mStoppedAtLocation == null
D/TSLocationManager( 3689): [c.t.l.l.TSLocationManager onLocationResult] 
D/TSLocationManager( 3689): ╔═════════════════════════════════════════════
D/TSLocationManager( 3689): ║ Process LocationResult
D/TSLocationManager( 3689): ╠═════════════════════════════════════════════
D/TSLocationManager( 3689): [c.t.l.l.TSLocationManager calculateMedianAccuracy] Median accuracy: 9.2
I/TSLocationManager( 3689): [c.t.l.data.sqlite.b persist] 
I/TSLocationManager( 3689):   ✅  INSERT: a5e0a862-b136-4dad-976a-4bc2690c3749
I/TSLocationManager( 3689): [c.t.l.http.HttpService flush] 
I/TSLocationManager( 3689): ╔═════════════════════════════════════════════
I/TSLocationManager( 3689): ║ HTTP Service (count: 1)
I/TSLocationManager( 3689): ╠═════════════════════════════════════════════
I/TSLocationManager( 3689): [c.t.l.u.BackgroundTaskManager onStartJob] ⏳ startBackgroundTask: 788
D/TSLocationManager( 3689): [c.t.l.data.sqlite.b allWithLocking] 
D/TSLocationManager( 3689):   ✅  Locked 1 records
I/TSLocationManager( 3689): [c.t.l.http.HttpService a] 
I/TSLocationManager( 3689):   ??????  HTTP POST batch (1)
I/TSLocationManager( 3689): [c.t.l.http.HttpService$i onResponse] 
I/TSLocationManager( 3689):   ??????  Response: 200
D/TSLocationManager( 3689): [c.t.l.data.sqlite.b destroyAll] 
D/TSLocationManager( 3689):   ✅  DELETED: (1)
D/TSLocationManager( 3689): [c.t.l.data.sqlite.b allWithLocking] 
D/TSLocationManager( 3689):   ✅  Locked 0 records
I/TSLocationManager( 3689): [c.t.l.u.BackgroundTaskManager$Task stop] ⏳ stopBackgroundTask: 788
D/TSLocationManager( 3689): [c.t.l.s.BackgroundTaskService onDestroy] 
```

</details>

**Comment by christocracy** [comment]
You need to gather far more logs than this, including logs that reproduce the issue.

See api docs BackgroundGeolocation.emailLog

**Comment by adrianaioanei** [comment]
I've attached the full logs.

Important events are:
16:40 start tracking
16:45 location icon goes off, no logs recorded anymore in adb logcat, lag starts to increase on the server
16:50 stop triggered manually, interestingly enough, a few location updates are recorded and sent to the server, then tracking again stops (obvious)

``` <!-- Syntax highlighting:  DO NOT REMOVE -->
╔═════════════════════════════════════════════
║ TSLocationManager version: 3.1.5 (361)
╠═════════════════════════════════════════════
╟─ Sony D6653 @ 5.0.2 (react-native)
{
  "activityRecognitionInterval": 0,
  "allowIdenticalLocations": true,
  "authorization": {
    "strategy": "JWT",
    "accessToken": "wsmLU<redacted>",
    "refreshPayload": {},
    "expires": -1
  },
  "autoSync": true,
  "autoSyncThreshold": 0,
  "batchSync": true,
  "configUrl": "",
  "debug": false,
  "deferTime": 0,
  "desiredAccuracy": -1,
  "desiredOdometerAccuracy": 100,
  "disableAutoSyncOnCellular": false,
  "disableElasticity": true,
  "disableLocationAuthorizationAlert": false,
  "disableMotionActivityUpdates": false,
  "disableStopDetection": true,
  "distanceFilter": 0,
  "elasticityMultiplier": 1,
  "enableHeadless": false,
  "enableTimestampMeta": false,
  "encrypt": false,
  "extras": {},
  "fastestLocationUpdateInterval": 333,
  "forceReloadOnBoot": false,
  "forceReloadOnGeofence": false,
  "forceReloadOnHeartbeat": false,
  "forceReloadOnLocationChange": false,
  "forceReloadOnMotionChange": false,
  "forceReloadOnSchedule": false,
  "foregroundService": true,
  "geofenceInitialTriggerEntry": true,
  "geofenceModeHighAccuracy": false,
  "geofenceProximityRadius": 1000,
  "geofenceTemplate": "",
  "headers": {},
  "headlessJobService": "com.transistorsoft.rnbackgroundgeolocation.HeadlessTask",
  "heartbeatInterval": 1,
  "httpRootProperty": "fixes",
  "httpTimeout": 60000,
  "isMoving": false,
  "locationAuthorizationRequest": "Always",
  "locationTemplate": "{\"latitude\": <%= latitude %>, \"longitude\": <%= longitude %>, \"timestamp-iso\": \"<%= timestamp %>\", \"speed\": <%= speed %>, \"course\": <%= heading %>, \"accuracy\": <%= accuracy %>}",
  "locationTimeout": 60,
  "locationUpdateInterval": 333,
  "locationsOrderDirection": "ASC",
  "logLevel": 5,
  "logMaxDays": 2,
  "maxBatchSize": 30,
  "maxDaysToPersist": 1,
  "maxRecordsToPersist": -1,
  "method": "POST",
  "minimumActivityRecognitionConfidence": 75,
  "motionTriggerDelay": 0,
  "notification": {
    "layout": "",
    "title": "",
    "text": "Event tracking activated",
    "color": "",
    "channelName": "TSLocationManager",
    "smallIcon": "",
    "largeIcon": "",
    "priority": 0,
    "strings": {},
    "actions": []
  },
  "params": {
    "deviceUuid": "c5ffc8a2-52a2-5761-b298-1c8e4b09a1b7"
  },
  "persist": true,
  "persistMode": 2,
  "schedule": [],
  "scheduleUseAlarmManager": false,
  "speedJumpFilter": 300,
  "startOnBoot": false,
  "stationaryRadius": 1,
  "stopAfterElapsedMinutes": 0,
  "stopOnStationary": false,
  "stopOnTerminate": true,
  "stopTimeout": 5,
  "triggerActivities": "in_vehicle, on_bicycle, on_foot, running, walking",
  "url": "https:\/\/my.sapsailing.com\/sailingserver\/api\/v1\/gps_fixes",
  "useSignificantChangesOnly": false,
  "enabled": false,
  "schedulerEnabled": false,
  "trackingMode": 1,
  "odometer": 10766.0537109375,
  "isFirstBoot": false,
  "didLaunchInBackground": false
}
╔═════════════════════════════════════════════
║ DEVICE SENSORS
╠═════════════════════════════════════════════
╟─ ✅  ACCELEROMETER: {Sensor name="BMA2X2 Accelerometer/Temperature/Double-tap", vendor="BOSCH", version=1, type=1, maxRange=39.226593, resolution=0.01914978, power=0.13, minDelay=5000}
╟─ ✅  GYROSCOPE: {Sensor name="BMG160 Gyroscope", vendor="BOSCH", version=1, type=4, maxRange=34.906586, resolution=0.0010681152, power=5.0, minDelay=5000}
╟─ ✅  MAGNETOMETER: {Sensor name="AK09911 Magnetometer", vendor="AKM", version=1, type=2, maxRange=4911.9995, resolution=0.5996704, power=0.24, minDelay=20000}
╟─ ✅  SIGNIFICANT_MOTION: {Sensor name="Significant Motion Detector", vendor="QTI", version=2, type=17, maxRange=1.0, resolution=1.0, power=0.12998962, minDelay=-1}
╚═════════════════════════════════════════════ 
```
[background-geolocation.log](https://github.com/transistorsoft/react-native-background-geolocation/files/4858801/background-geolocation.log)

**Comment by christocracy** [comment]
Is this from real-world field-testing or a Mock location app?

**Comment by adrianaioanei** [comment]
Real-world, no mocks, but from a device staying still on a desk all the time.

**Comment by christocracy** [comment]
Restart the device and try again.

**Comment by christocracy** [comment]
- Plugin determines your app has entered background.  Fires a one-shot timer in 10s in order to determine if app is terminated.
```
07-01 16:45:48.237 INFO [TSScheduleManager oneShot] 
  ⏰ Scheduled OneShot: TERMINATE_EVENT in 10000ms (jobID: -1708771588)
```

- 10s later, the one-shot timer elapses.  Your app is found to *not* be terminated.
07-01 16:45:58.243 INFO [ScheduleEvent onOneShot] 
╔═════════════════════════════════════════════
║ ⏰ OneShot event fired: TERMINATE_EVENT
╠═════════════════════════════════════════════

07-01 16:45:58.246 DEBUG [TerminateEvent <init>] 
  ℹ️  TERMINATE_EVENT ignored (MainActivity is still active).
```

- Your app is determined to have entered background *again*.  
07-01 16:49:17.382 INFO [TSScheduleManager oneShot] 
  ⏰ Scheduled OneShot: TERMINATE_EVENT in 10000ms (jobID: -1708771588)
```

- Once again, 10s later, the oneshot fires again.  Your app is still *not* terminated.
```
07-01 16:49:27.396 INFO [ScheduleEvent onOneShot] 
╔═════════════════════════════════════════════
║ ⏰ OneShot event fired: TERMINATE_EVENT
╠═════════════════════════════════════════════

07-01 16:49:27.405 DEBUG [TerminateEvent <init>] 
  ℹ️  TERMINATE_EVENT ignored (MainActivity is still active).
```

- Something sets `isMoving: false`.  The plugin enters the *stationary* state and turns off location-services.
```
07-01 16:50:05.427 DEBUG [TSConfig d] ℹ️   Persist config, dirty: [isMoving]
07-01 16:50:05.437 INFO [TSLocationManager removeLocationUpdates] 
  🔴  Location-services: OFF
07-01 16:50:05.441 INFO [TSGeofenceManager stopMonitoringSignificantLocationChanges] 
  🔴  Stop monitoring significant location changes
07-01 16:50:05.448 INFO [HeartbeatService b] 
  🎾  Start heartbeat (1s)
07-01 16:50:05.458 DEBUG [b b] 
  ℹ️  LocationAuthorization: Permission granted
07-01 16:50:05.459 INFO [TrackingService a] 
  🔵  setPace: true → false
```

**Comment by christocracy** [comment]
I'd be checking if you're doing anything in the app life-cycle pause/resume events, such as calling `#ready` / `#setConfig`

**Comment by adrianaioanei** [comment]
The OneShot events came after tracking stopped, as I closed and reopened the app to see if it would change anything.
Then at 16:50 i manually invoked `changePage(false)` then `stop`. But that is 5 minutes after tracking had stopped, at 16:45.

**Comment by christocracy** [comment]
The last recorded location is `16:45`.  Location-services don't turn off until `16:50`, *after* your app goes to background.
```
07-01 16:45:24.862 DEBUG [TrackingService b] 
╔═════════════════════════════════════════════
║ TrackingService: LocationResult
╠═════════════════════════════════════════════
╟─ 📍  Location[fused 54.430038,10.168422 acc=10 et=+1d2h13m32s796ms alt=47.69999694824219 vel=1.5105617E-4 bear=180.0 {Bundle[mParcelledData.dataSize=52]}], age: 99ms, time: 1593614725000
.
.
.
╔═════════════════════════════════════════════
║ ⏰ OneShot event fired: TERMINATE_EVENT
╠═════════════════════════════════════════════

07-01 16:49:27.405 DEBUG [TerminateEvent <init>] 
  ℹ️  TERMINATE_EVENT ignored (MainActivity is still active).
07-01 16:50:05.427 DEBUG [TSConfig d] ℹ️   Persist config, dirty: [isMoving]
07-01 16:50:05.437 INFO [TSLocationManager removeLocationUpdates] 
  🔴  Location-services: OFF
```

Location-services goes OFF right after something caused after `isMoving` changed to `false`

**Comment by adrianaioanei** [comment]
Correct, that aligns with my findings as well. Starting with 16:45, there was no location sent to the server.
What happens starting with 16:49 is straightforward and makes sense because I sent the app to the background and then stopped tracking manually.
But why, all of a sudden, it would stop sending location at 16:45?

**Comment by christocracy** [comment]
>But why, all of a sudden, it would stop sending location at 16:45

Perhaps you went indoors at that point, losing the GPS signal?  The device's Location-services API was probably busy attempting to reconnect with GPS satellites / looking for Wifi signals.

**Comment by adrianaioanei** [comment]
No, the phone was always staying still on a desk in the office. No change in environment during the session.

**Comment by adrianaioanei** [comment]
> Restart the device and try again.

I got to do another round of testing on the same device. Same behavior. After 5 minutes, tracking stops without the user doing anything. Again with the phone always staying still on the desk. Attached are the logs.
I started tracking at 21.31, it stopped sending locations at around 21.37.06.
Then at 21.37.43 I manually invoked stop on the module. 
[background-geolocation (1).log](https://github.com/transistorsoft/react-native-background-geolocation/files/4881158/background-geolocation.1.log)
